src/components/FileUpload/FileUpload.js

```diff
@@ -0,0 +1,175 @@
+import Image from 'next/image';
+import { useState } from 'react';
+import { StyledFileUpload, StyledInput } from './StyledFileUpload';
+import StyledFlex from '../../../styles/StyledFlex';
+import StyledButton from '../../../styles/StyledButton';
+import _useStore from '../../lib/hooks/_useStore';
+import SVGIcons from '../../assets/SVGIcon/SVGIcons';
+import useStore from '../../lib/hooks/useStore';
+
+export default function FileUpload() {
+	const CLOUD = process.env.CLOUDINARY_CLOUD;
+	const PRESET = process.env.CLOUDINARY_PRESET_DOCUMENTS;
+
+	const toggleAddDocument = _useStore(state => state.toggleAddDocument);
+
+	const addDocument = useStore(state => state.addDocument);
+	const id = useStore(state => state.id);
+
+	const [previewDocument, setPreviewDocument] = useState();
+	const [documentTitel, setDocumentTitel] = useState('');
+
+	const uploadDocument = async event => {
+		try {
+			const url = `https://api.cloudinary.com/v1_1/${CLOUD}/image/upload`;
+			const file = event.target.files[0];
+
+			const data = new FormData();
+			data.append('file', file);
+			data.append('upload_preset', PRESET);
+
+			const response = await fetch(url, {
+				method: 'POST',
+				body: data,
+			});
+
+			setPreviewDocument(await response.json());
+		} catch (error) {
+			console.error(error.message);
+		}
+	};
+
+	const onSubmit = data => {
+		data.file = {
+			url: previewDocument.url,
+			width: previewDocument.width,
+			height: previewDocument.height,
+		};
+		console.log(data);
+	};
+
+	return (
+		<>
+			<StyledFlex
+				height="65vh"
+				background="var(--secondary-color)"
+				justifyContent="center"
+				alignSelf="center"
+				borderRadius="8px"
+				position="fixed"
+				top="0"
+				right="0"
```

Comment:
Würde man bei so vielen Styling Änderrungen nicht eher eine neue Style Component erstellen?

Reply:
Jup, wie besprochen bei Zeiten ein Refactoring zum Styling und das komponentenspezifische Styling in die jeweilige Komponente umzuziehen :)